docs/v0.5/getting-started/kind.md

```diff
@@ -0,0 +1,307 @@
+
+```sh
+riff function create square \
+  --git-repo https://github.com/projectriff-samples/node-square  \
+  --artifact square.js \
+  --tail
+```
+
+After the function is created, you can see the built image by listing functions.
+
+```sh
+riff function list
+```
+
+```
+NAME     LATEST IMAGE                                                                                                ARTIFACT    HANDLER   INVOKER   STATUS   AGE
+square   index.docker.io/$DOCKER_ID/square@sha256:ac089ca183368aa831597f94a2dbb462a157ccf7bbe0f3868294e15a24308f68   square.js   <empty>   <empty>   Ready    1m13s
+```
+
+## Create a Knative deployer
+
+The [Knative Runtime](../runtimes/knative.md) is only available on clusters with Istio and Knative installed. Knative deployers run riff workloads using Knative resources which provide auto-scaling (including scale-to-zero) based on HTTP request traffic, and routing.
+
+```sh
+riff knative deployer create knative-square --function-ref square --tail
+```
+
+After the deployer is created, you can see the hostname by listing deployers.
+
+```sh
+riff knative deployer list
+```
+```
+NAME             TYPE       REF      HOST                                 STATUS   AGE
+knative-square   function   square   knative-square.default.example.com   Ready    28s
+```
+
+### invoke the function
+
+Knative configures HTTP routes on the istio-ingressgateway. Requests are routed by hostname.
+
+Look up the IP address for the kind node; you should see an address like `172.17.0.3`.
+
+```sh
+KIND_NODE_IP=$(kubectl get node -o jsonpath='{$.items[0].status.addresses[?(@.type=="InternalIP")].address}')
+echo $KIND_NODE_IP
+```
+
+Look up the nodePort for the ingressgateway; you should see a port value like `30195`.
+
+```sh
+INGRESS_PORT=$(kubectl get svc istio-ingressgateway --namespace istio-system --output 'jsonpath={.spec.ports[?(@.port==80)].nodePort}')
+echo $INGRESS_PORT
+```
+
+Invoke the function by POSTing to the ingressgateway, passing the hostname and content-type as headers.
+
+```sh
+curl http://$KIND_NODE_IP:$INGRESS_PORT/ -w '\n' \
+-H 'Host: knative-square.default.example.com' \
+-H 'Content-Type: application/json' \
+-d 7
+```
+```
+49
+```
+
+## Create a Core deployer
+
+The [Core runtime](../runtimes/core.md) deploys riff workloads as "vanilla" Kubernetes deployments and services.
+
+```sh
+riff core deployer create k8s-square --function-ref square --tail
+```
+
+After the deployer is created, you can see the service name by listing deployers.
+
+```sh
+riff core deployers list
+```
+```
+NAME         TYPE       REF      SERVICE               STATUS   AGE
+k8s-square   function   square   k8s-square-deployer   Ready    16s
+```
+
+### invoke the function
+
+In a separate terminal, start port-forwarding to the ClusterIP service created by the deployer.
+
+```sh
+kubectl port-forward service/k8s-square-deployer 8080:80
+```
```

Comment:
we need to look up the generated service name, this worked:
```
kubectl port-forward $(kubectl get svc -oname -l core.projectriff.io/deployer=k8s-square) 8080:80
```

Reply:
newer riff runtimes have ingress